File: stock-market-service/src/main/java/rs/edu/raf/repository/OrderRepository.java
package rs.edu.raf.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import rs.edu.raf.model.order.Order;
import rs.edu.raf.model.order.OrderAction;

import java.util.List;

@Repository
public interface OrderRepository extends JpaRepository<Order, Long> {
    List<Order> findAllByTickerAndOrderActionAndQuantityGreaterThan(String ticker, OrderAction orderAction, int quantity);
    List<Order> findAllByQuantityGreaterThan(int quantity);
}

File: stock-market-service-kotlin/src/main/kotlin/org/example/stockmarketservicekotlin/repository/OrderRepository.kt
package org.example.stockmarketservicekotlin.repository

import org.example.stockmarketservicekotlin.model.order.Order
import org.example.stockmarketservicekotlin.model.order.OrderAction
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface OrderRepository : JpaRepository<Order, Long> {
    fun findAllByTickerAndOrderActionAndQuantityGreaterThan(
        ticker: String, orderAction: OrderAction, quantity: Int
    ): List<Order>

    fun findAllByQuantityGreaterThan(quantity: Int): List<Order>
}